src/backend/distributed/utils/node_metadata.c

```diff
@@ -345,6 +346,7 @@ AddNodeMetadata(char *nodeName, int32 nodePort, int32 groupId, char *nodeRack,
 	char *nodeInsertCommand = NULL;
 	List *workerNodeList = NIL;
 
+	EnsureSchemaNode();
```

Comment:
Minor note, you could decide what to do:
In other places, we error out on the UDF definition itself. For `master_add_node` and `master_initialize_node_metadata` we use this function to error out and make it less explicit compared to others. Plus, the other way would make code more consistent.
(and similar pattern for `create_reference_table`)